src/template_parser.rs

```diff
@@ -94,31 +95,59 @@ impl TemplateProperty<Signature, String> for SignatureEmail {
     }
 }
 
+fn datetime_from_signature(context: &Signature) -> Option<DateTime<FixedOffset>> {
+    let utc = match Utc.timestamp_opt(
+        context.timestamp.timestamp.0.div_euclid(1000),
+        (context.timestamp.timestamp.0.rem_euclid(1000)) as u32 * 1000000,
+    ) {
+        LocalResult::None => {
+            return None;
+        }
+        LocalResult::Single(x) => x,
+        LocalResult::Ambiguous(y, _z) => y,
+    };
+
+    Some(
+        utc.with_timezone(
+            &FixedOffset::east_opt(context.timestamp.tz_offset * 60)
+                .unwrap_or_else(|| FixedOffset::east_opt(0).unwrap()),
+        ),
+    )
+}
+
 struct SignatureTimestamp;
 
 impl TemplateProperty<Signature, String> for SignatureTimestamp {
     fn extract(&self, context: &Signature) -> String {
-        let utc = match Utc.timestamp_opt(
-            context.timestamp.timestamp.0.div_euclid(1000),
-            (context.timestamp.timestamp.0.rem_euclid(1000)) as u32 * 1000000,
-        ) {
-            LocalResult::None => {
-                return "<out-of-range date>".to_string();
-            }
-            LocalResult::Single(x) => x,
-            LocalResult::Ambiguous(y, _z) => y,
-        };
-        let datetime = utc.with_timezone(
-            &FixedOffset::east_opt(context.timestamp.tz_offset * 60)
-                .unwrap_or_else(|| FixedOffset::east_opt(0).unwrap()),
-        );
-        datetime.format("%Y-%m-%d %H:%M:%S.%3f %:z").to_string()
+        match datetime_from_signature(context) {
+            Some(datetime) => datetime.format("%Y-%m-%d %H:%M:%S.%3f %:z").to_string(),
+            None => "<out-of-range date>".to_string(),
+        }
+    }
+}
+
+struct SignatureTimestampHumanized;
+
+impl TemplateProperty<Signature, String> for SignatureTimestampHumanized {
+    fn extract(&self, context: &Signature) -> String {
+        datetime_from_signature(context)
+            .and_then(|datetime| {
+                let now = chrono::Local::now();
+
+                now.signed_duration_since(datetime).to_std().ok()
+            })
+            .map(|duration| {
+                let f = timeago::Formatter::new();
+                f.convert(duration)
+            })
+            .unwrap_or_else(|| "<out-of-range date>".to_string())
     }
 }
 
 fn parse_method_chain<'a, I: 'a>(
     pair: Pair<Rule>,
     input_property: Property<'a, I>,
+    settings: &UserSettings,
```

Comment:
Related your comment about "It's a bit awkward having this be a setting since it requires passing the settings throughout the parsing code.":

I think it would be more flexible to instead keep the current behavior where `author.timestamp()` prints the absolute timestamp, and to make `author.timestamp().ago()` print a duration. Then we can make the user's config change the default template to produce a template with `.ago()` in.

To do that, I think you would do this:

* Add a `SignatureTimestamp` in `templater.rs` and make it implement `TemplateProperty<Signature, Timestamp>`
* Replace `SignatureTimestamp` in `template_parser.rs` by a `TimestampString` and make it implement `TemplateProperty<Timestamp, String>`
* Add a `Timestamp` variant to the `Property` enum
* Probably many more steps :) Let me know if you need help.

Reply:
That makes total sense and is much cleaner, updated as mentioned.